### ISSUE TITLE
Is it wrong for tf.reduce_sum(tf.square(ddx), axis=1)?

### ISSUE THREAD

**Issue by uuutty** [jiamings/wgan#9]
@jiamings  In your `wgan-v2.py`, you write 
`ddx = tf.sqrt(tf.reduce_sum(tf.square(ddx), axis=1))`
this will result a shape `(50, 64, 64, 3)` to `(50, 64, 3)` in tensorflow 1.6.0. 
If I understand right for the paper [_Improved Training of Wasserstein GANs_](https://arxiv.org/abs/1704.00028), this should be
`ddx = tf.sqrt(tf.reduce_sum(tf.square(ddx), axis=[1,2,3]))` which get a shape of `(50,)` indicate the batch size.
